### PR TITLE
feat(apollo): makes optional the enable of the landing page playground

### PR DESCRIPTION
**Pull request by tehKapa**
Related to https://github.com/sorry-cypress/sorry-cypress/pull/624

### CODE REVIEW

**Comment by tico24** [comment]
Hi there. Thanks for your submission.

The readme, changelog and chart version will need to be updated to reflect the change.
You've done more than just make the landing page optional. I'd prefer if your formatting changes were in a different PR.

**Comment by tehKapa** [comment]
@tico24 I've updated the changelog and the Charts version with another commit. I don't understand what you mean by " I'd prefer if your formatting changes were in a different PR."

**Comment by tico24** [comment]
You've made other changes to the values.yaml that don't relate to this feature.

**Comment by tehKapa** [comment]
> You've made other changes to the values.yaml that don't relate to this feature.

@tico24 gotcha. fixed

**Comment by tico24** [comment]
Just the merge conflicts to go I'm afraid.

**Comment by tehKapa** [comment]
@tico24 done!

**Comment by tico24** [comment]
@tehKapa You know this hasn't been merged in because the CI keeps failing yeah?

**Comment by tehKapa** [comment]
@tico24 I hadn't noticed that. Thank you

**Comment by tehKapa** [comment]
@tico24 now it should pass lint step

```
➜ ct lint --all --debug --config lint.yaml
Linting charts...
Version increment checking disabled.
>>> helm version --short

------------------------------------------------------------------------------------------------------------------------
 Charts to be processed:
------------------------------------------------------------------------------------------------------------------------
 sorry-cypress => (version: "1.8.0", path: "charts/sorry-cypress")
------------------------------------------------------------------------------------------------------------------------

>>> helm repo add minio https://helm.min.io
"minio" already exists with the same configuration, skipping
>>> helm repo add mongodb https://raw.githubusercontent.com/bitnami/charts/pre-2022/bitnami
"mongodb" already exists with the same configuration, skipping
>>> helm dependency build charts/sorry-cypress
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "minio" chart repository
...Successfully got an update from the "mongodb" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 2 charts
Downloading minio from repo https://helm.min.io/
Downloading mongodb from repo https://raw.githubusercontent.com/bitnami/charts/pre-2022/bitnami
Deleting outdated charts
Linting chart 'sorry-cypress => (version: "1.8.0", path: "charts/sorry-cypress")'
>>> yamale --schema /usr/local/etc/ct/chart_schema.yaml charts/sorry-cypress/Chart.yaml
Validating /Volumes/ExternalSSD/Dev/sorry-cypress/charts/charts/sorry-cypress/Chart.yaml...
Validation success! 👍
>>> yamllint --config-file /usr/local/etc/ct/lintconf.yaml charts/sorry-cypress/Chart.yaml
>>> yamllint --config-file /usr/local/etc/ct/lintconf.yaml charts/sorry-cypress/values.yaml
>>> yamllint --config-file /usr/local/etc/ct/lintconf.yaml charts/sorry-cypress/ci/isolated-director-values.yaml
>>> yamllint --config-file /usr/local/etc/ct/lintconf.yaml charts/sorry-cypress/ci/latest-image-values.yaml
>>> yamllint --config-file /usr/local/etc/ct/lintconf.yaml charts/sorry-cypress/ci/minio-values.yaml
>>> yamllint --config-file /usr/local/etc/ct/lintconf.yaml charts/sorry-cypress/ci/mongo-connectionstring-values.yaml
>>> yamllint --config-file /usr/local/etc/ct/lintconf.yaml charts/sorry-cypress/ci/mongo-secret-connectionstring-values.yaml
>>> yamllint --config-file /usr/local/etc/ct/lintconf.yaml charts/sorry-cypress/ci/mongo-values.yaml
>>> yamllint --config-file /usr/local/etc/ct/lintconf.yaml charts/sorry-cypress/ci/multiple-replicas-values.yaml
Validating maintainers...
>>> git ls-remote --get-url origin

Linting chart with values file 'charts/sorry-cypress/ci/isolated-director-values.yaml'...

>>> helm lint charts/sorry-cypress --values charts/sorry-cypress/ci/isolated-director-values.yaml
==> Linting charts/sorry-cypress

1 chart(s) linted, 0 chart(s) failed

Linting chart with values file 'charts/sorry-cypress/ci/latest-image-values.yaml'...

>>> helm lint charts/sorry-cypress --values charts/sorry-cypress/ci/latest-image-values.yaml
==> Linting charts/sorry-cypress

1 chart(s) linted, 0 chart(s) failed

Linting chart with values file 'charts/sorry-cypress/ci/minio-values.yaml'...

>>> helm lint charts/sorry-cypress --values charts/sorry-cypress/ci/minio-values.yaml
==> Linting charts/sorry-cypress

1 chart(s) linted, 0 chart(s) failed

Linting chart with values file 'charts/sorry-cypress/ci/mongo-connectionstring-values.yaml'...

>>> helm lint charts/sorry-cypress --values charts/sorry-cypress/ci/mongo-connectionstring-values.yaml
==> Linting charts/sorry-cypress

1 chart(s) linted, 0 chart(s) failed

Linting chart with values file 'charts/sorry-cypress/ci/mongo-secret-connectionstring-values.yaml'...

>>> helm lint charts/sorry-cypress --values charts/sorry-cypress/ci/mongo-secret-connectionstring-values.yaml
==> Linting charts/sorry-cypress

1 chart(s) linted, 0 chart(s) failed

Linting chart with values file 'charts/sorry-cypress/ci/mongo-values.yaml'...

>>> helm lint charts/sorry-cypress --values charts/sorry-cypress/ci/mongo-values.yaml
==> Linting charts/sorry-cypress

1 chart(s) linted, 0 chart(s) failed

Linting chart with values file 'charts/sorry-cypress/ci/multiple-replicas-values.yaml'...

>>> helm lint charts/sorry-cypress --values charts/sorry-cypress/ci/multiple-replicas-values.yaml
==> Linting charts/sorry-cypress

1 chart(s) linted, 0 chart(s) failed
------------------------------------------------------------------------------------------------------------------------
 ✔︎ sorry-cypress => (version: "1.8.0", path: "charts/sorry-cypress")
------------------------------------------------------------------------------------------------------------------------
All charts linted successfully
```

File: charts/sorry-cypress/templates/deployment-api.yml [hunk 1]
```diff
@@ -48,6 +48,8 @@ spec:
         {{-  $defaultConnectionSecretName := include "sorry-cypress-helm.fullname" . | printf "%s-mongodbsecrets" }}
         {{-  $defaultConnectionSecretKey := "MONGODB_URI" }}
         {{- if .Values.mongodb.mongoSecretConnectionString.enableCustomSecret }}
```
Comment (tico24) [inline charts/sorry-cypress/templates/deployment-api.yml:50]: Why is this nested within the mongo if?

**Review by tico24 — CHANGES_REQUESTED** [review]

File: charts/sorry-cypress/templates/deployment-api.yml (hunk 1 above)
Comment (tehKapa) [reply charts/sorry-cypress/templates/deployment-api.yml]: something was wrong during a merge conflict, I guess

File: charts/sorry-cypress/templates/deployment-api.yml [hunk 2]
```diff
@@ -63,6 +63,8 @@ spec:
           value: "mongodb://{{ include "mongodb.hostname" . }}:{{ .Values.mongodb.service.port }}"
         {{- end }}
         {{- end }}
+        - name: APOLLO_PLAYGROUND
+          value: {{ .Values.api.environmentVariables.apolloPlayground}}
```
Comment (tico24) [inline charts/sorry-cypress/templates/deployment-api.yml:67]: missing space before the `}}` I suspect this is what's breaking the CI.

Strongly recommend testing locally before pushing.

**Review by tico24 — CHANGES_REQUESTED** [review]

File: charts/sorry-cypress/templates/deployment-api.yml (hunk 2 above)
Comment (tehKapa) [reply charts/sorry-cypress/templates/deployment-api.yml]: Done

**Comment by tehKapa** [comment]
@tico24 I cannot understand how and why it failed. I'll try to reproduce this error with ACT on my local machine

**Comment by tehKapa** [comment]
Ok thanks!

**Pull request closed without merging**